common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -164,8 +166,10 @@ def make_selection(cls, selected, children, max_count, mode):
 
         # If max_count has been decreased, we may have to drop some previously selected blocks:
         overlimit_block_keys = set()
-        while len(selected) > max_count:
-            overlimit_block_keys.add(selected.pop())
+        if len(selected) > max_count:
+            num_to_remove = len(selected) - max_count
+            overlimit_block_keys = set(rand.sample(selected, num_to_remove))
+            selected -= overlimit_block_keys
```

Comment:
Does `selected` get saved somewhere for this user - so they get the same `selected` values the next time around?

Reply:
Yup, this is called from from `selected_children()` in the same class and stored in `self.selected` (it's also cached in `self._selected_set`).